ansible/playbooks/aix.yml

```diff
@@ -6,125 +6,278 @@
 - hosts: all
   remote_user: root
   become: yes
+  environment:
+    PATH: "/opt/freeware/bin:{{ ansible_env.PATH }}"
+  vars:
+    swap_size: 4096
 
   tasks:
   - block:
-
-    ##########
-    # Layout #
-    ##########
-    - name: Create tmp folder /tmp/playbook_tmp
-      file: path=/tmp/playbook_tmp state=directory mode=0755
-      tags: layout
-
-    - name: Create rpm tmp folder /tmp/playbook_tmp/rpms
-      file: path=/tmp/playbook_tmp/rpms state=directory mode=0755
-      tags: layout
-
-    - name: Create java tmp folder /tmp/playbook_tmp/java
-      file: path=/tmp/playbook_tmp/java state=directory mode=0755
-      tags: layout
-
-    - name: Create xlc tmp folder /tmp/playbook_tmp/xlc
-      file: path=/tmp/playbook_tmp/xlc state=directory mode=0755
-      tags: layout
-
-    ################################
-    # AIX filesystem configuration #
-    ################################
+  ################################
+  # AIX filesystem configuration #
+  ################################
     - name: Download AIX filesystem configuration script
       get_url:
         url: https://raw.githubusercontent.com/AdoptOpenJDK/openjdk-infrastructure/master/ansible/playbooks/scripts/AIX_filesystem_config.sh
-        dest: /tmp/playbook_tmp/AIX_filesystem_config.sh
+        dest: /tmp/AIX_filesystem_config.sh
         mode: 0775
         validate_certs: no
       tags: filesystem
         
     - name: Execute AIX filesystem configuration script
-      command: /tmp/playbook_tmp/AIX_filesystem_config.sh
+      command: /tmp/AIX_filesystem_config.sh
       tags: filesystem
 
-    ######################################
-    # Install and configure RPM Packages #
-    ######################################
-    - name: Transfer RPM Packages to remote host
+    - file:
+        state: absent
+        path: /tmp/AIX_filesystem_config.sh
+      tags: filesystem
+
+  #####################################
+  # Install openssl from IBM installp #
+  #####################################
+    - name: Transfer Openssl to remote host
       copy:
-        src: "{{ item }}"
-        dest: /tmp/playbook_tmp/rpms/
-        mode: 0755
-      with_fileglob:
-        - /home/ubuntu/aix/rpms/*
-      tags: additional_rpms
+        src: /Vendor_Files/aix/openssl-1.0.2.1300.tar.gz
+        dest: /tmp/openssl-1.0.2.1300.tar.gz
+      tags: openssl
+
+    - name: Extract Openssl 
+      shell: gzip -d /tmp/openssl-1.0.2.1300.tar.gz
+      tags: openssl
+    
+    - shell: tar -xvf /tmp/openssl-1.0.2.1300.tar -C /tmp/
+      tags: openssl
+
```

Comment:
The previous 3 steps could be done in one "uncompress", like I do in other spots in the file, but there is no unzip installed by default, and its not installed until the yum step(s).

Reply:
Not sure why `unzip` is relevant here: This should do it for the last two steps at least: `gzip -cd /tmp/openssl-1.0.2.1300.tar.gz | tar xf - -C /tmp/`

Reply:
The unarchive method can transfer the file and unpack in one step instead of 3 (or 2 with your proposed change) but it depends on unzip.  I was simply meaning that I couldn't use the ansible preferred method as the warning when you run it in AWS states "Consider using unarchive module rather than running tar".  I simplified with your suggestion.